### ISSUE TITLE
Upstream to our main 

### ISSUE THREAD

**Issue by mounakrishna99** [Mindgrove-Technologies/mindgrove-bsc#12]
Bringing in upstream bluespec changes to our fork.

**Comment by mounakrishna99** [comment]
@shash @SharanJ @RohitSrinivasRG @ssanthoshpavan 
I have merged the upstream bluespec with the development branch. There are changes in many, but none affects SecureIoT as of now.

I generated verilog with both old and new bsc, both are similar with just changes in variable names.

My suggestion, once secureIoT production is done, we will tag the current main branch bsc as SecureIoT production BSC and then merge development to our main and continue with that for vision-soc. Is this fine?

**Comment by SharanJ** [comment]
Yeah. Sounds good. I think we should spend some time later to contribute back to the upstream repo. What do you think @shash  and @mounakrishna99 ?

**Comment by shash** [comment]
The intent is to contribute back. Sent from my iPhoneOn 20 Nov 2024, at 08:23, Sharan Jagathrakshakan ***@***.***> wrote:﻿
Yeah. Sounds good. I think we should spend some time later to contribute back to the upstream repo. What do you think @shash  and @mounakrishna99 ?
